From 3ca6e1cacb80ca8e6ad2abbd6ba3354f1ed9eb24 Mon Sep 17 00:00:00 2001
From: Darren-Hill <Darren-Hill@users.noreply.github.com>
Date: Sat, 11 May 2019 16:18:52 +0100
Subject: [PATCH 1/2] Blade menu access for touchscreens

Adds two-finger right swipe for access to the blade menu on touchscreens.
---
 system/keymaps/touchscreen.xml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/system/keymaps/touchscreen.xml b/system/keymaps/touchscreen.xml
index 1d5903564564..56a42c842cb2 100644
--- a/system/keymaps/touchscreen.xml
+++ b/system/keymaps/touchscreen.xml
@@ -30,6 +30,7 @@
       <swipe direction="left">SwipeLeft</swipe>
       <swipe direction="left" pointers="2">Back</swipe>
       <swipe direction="right">SwipeRight</swipe>
+      <swipe direction="right" pointers="2">Menu</swipe>  <!-- Easier access to the side blade menu on touchscreens.  -->
       <swipe direction="up">SwipeUp</swipe>
       <swipe direction="down">SwipeDown</swipe>
       <swipe direction="up" pointers="3">SwitchPlayer</swipe>

From 950016066a0defdfc470753eb359edb3e81c0273 Mon Sep 17 00:00:00 2001
From: Darren-Hill <Darren-Hill@users.noreply.github.com>
Date: Sat, 11 May 2019 16:21:06 +0100
Subject: [PATCH 2/2] Blade menu and context menu exit

Adds blade menu access through long right-click.
Adds right click as a way to exit the context menu without selecting anything.
---
 system/keymaps/mouse.xml | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/system/keymaps/mouse.xml b/system/keymaps/mouse.xml
index b4edcd482d2a..85b184d769ff 100644
--- a/system/keymaps/mouse.xml
+++ b/system/keymaps/mouse.xml
@@ -34,6 +34,7 @@
       <middleclick>middleclick</middleclick>
       <doubleclick id="0">doubleclick</doubleclick>
       <longclick id="0">contextmenu</longclick>
+      <longclick id="2">Menu</longclick>  <!-- access to the blade menu via long right-click -->
       <wheeldown>wheeldown</wheeldown>
       <wheelup>wheelup</wheelup>
       <mousedrag>mousedrag</mousedrag>
@@ -58,4 +59,9 @@
       <rightclick>Info</rightclick>
     </mouse>
   </FullscreenGame>
+  <contextmenu>  <!-- Give a way out of the context menu without actually having to select something.  -->
+    <mouse>
+      <rightclick>back</rightclick>
+    </mouse>
+  </contextmenu>
 </keymap>
